src/config.rs

```diff
@@ -1066,7 +1069,7 @@ fn config_overrides() {
 #[test]
 #[serial]
 fn test_s3_no_credentials() {
-    env::set_var("SCCACHE_S3_NO_CREDENTIALS", "1");
+    env::set_var("SCCACHE_S3_NO_CREDENTIALS", "true");
```

Comment:
we would need more (and separate) tests to cover this

Reply:
Updated in the latest commits. I mostly just copy and pasted the existing test functions.

Let me know if there's a better convention than that.